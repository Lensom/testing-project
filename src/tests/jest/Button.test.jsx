import React from 'react';
import { describe, expect, test } from '@jest/globals';
import { render, screen, fireEvent } from '@testing-library/react';
import CustomButton from '../../components/Button';

describe('Testing Button component', () => {
  test('Click on button', async () => {
    const { container } = render(<CustomButton />);

    const button = screen.getByText('Plus');
    const counter = container.querySelector('p');

    expect(counter.textContent).toBe('Counter: 0');

    fireEvent.click(button);

    expect(counter.textContent).toBe('Counter: 1');
  });

  test('Matches snapshot', () => {
    const { container } = render(<CustomButton />);

    expect(container).toMatchSnapshot();
  });

  test('Sending prop day to compononent', async () => {
    const { container } = render(<CustomButton day={'01.01.2025'} />);
    const currentDay = container.querySelector('span');

    expect(currentDay.textContent).toBe('Current day: 01.01.2025');
  });
});
