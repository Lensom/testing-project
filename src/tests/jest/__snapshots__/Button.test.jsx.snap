// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Button component Matches snapshot 1`] = `
<div>
  <button>
    Plus
  </button>
  <p>
    Counter: 
    0
  </p>
  <span>
    Current day: 
  </span>
</div>
`;
